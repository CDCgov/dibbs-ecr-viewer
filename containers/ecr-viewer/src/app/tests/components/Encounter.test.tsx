import { render } from "@testing-library/react";
import { axe } from "jest-axe";
import EncounterDetails from "../../view-data/components/Encounter";
import React from "react";

describe("Encounter", () => {
  let container: HTMLElement;
  beforeAll(() => {
    const mockChildMethod = jest.fn();
    // This gets the dynamic tooltip IDs to show up as stable "[object Object]"
    jest.spyOn(React, "useRef").mockReturnValue({
      current: {
        childMethod: mockChildMethod,
      },
    });
    const encounterData = [
      {
        title: "Encounter Type",
        value: "Ambulatory",
      },
      {
        title: "Encounter ID",
        value: "123456789",
      },
      {
        title: "Encounter Care Team",
        table: true,
        value: (
          <table>
            <thead>
              <tr>
                <th>Name</th>
                <th>Role</th>
                <th>Dates</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>Test</td>
                <td>ATND</td>
                <td>Start: 1/2/2023</td>
              </tr>
            </tbody>
          </table>
        ),
      },
    ];
    const facilityData = [
      {
        title: "Facility Name",
        value: "PRM- Palmdale Regional Medical Center",
      },
      {
        title: "Facility Address",
        value:
          "5001 North Mount Washington Circle Drive\nNorth Canton, MA 02740",
      },
      {
        title: "Facility Contact Address",
        value:
          "5001 North Mount Washington Circle Drive\nNorth Canton, MA 02740",
      },
      {
        title: "Facility Type",
        value: "Healthcare Provider",
      },
      {
        title: "Facility ID",
        value: "2.16.840.1.113883.4.6",
      },
    ];
    const providerData = [
      {
        title: "Provider Name",
        value: "test provider name",
      },
      {
        title: "Provider Contact",
        value: "test provider contact",
      },
    ];
    container = render(
      <EncounterDetails
        encounterData={encounterData}
        facilityData={facilityData}
        providerData={providerData}
      />,
    ).container;
  });
  it("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });
  it("should pass accessibility test", async () => {
    expect(await axe(container)).toHaveNoViolations();
  });
});
