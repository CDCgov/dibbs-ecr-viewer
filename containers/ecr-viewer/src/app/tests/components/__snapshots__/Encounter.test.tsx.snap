// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Encounter should match snapshot 1`] = `
<div>
  <div
    class="margin-top-0"
  >
    <div
      class="padding-bottom-3"
    >
      <div
        class="usa-summary-box__body"
      >
        <div
          class="gutter-3"
        >
          <h4
            class="usa-summary-box__heading padding-y-105"
            id="encounter-details"
          >
            Encounter Details
          </h4>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Encounter Type
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  Ambulatory
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Encounter ID
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  123456789
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div
              class="grid-row"
            >
              <div
                class="grid-col-auto width-full text-pre-line"
              >
                <table>
                  <thead>
                    <tr>
                      <th>
                        Name
                      </th>
                      <th>
                        Role
                      </th>
                      <th>
                        Dates
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td>
                        Test
                      </td>
                      <td>
                        ATND
                      </td>
                      <td>
                        Start: 1/2/2023
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
            <div
              class="section__line_gray"
            />
          </div>
        </div>
        <div
          class="gutter-3"
        >
          <span
            class="usa-tooltip"
            data-testid="tooltipWrapper"
          >
            <div
              aria-describedby="[object Object]"
              class="usa-tooltip__trigger usa-tooltip short-tooltip"
              data-testid="triggerElement"
              tabindex="0"
              title=""
            >
              <h4
                class="usa-summary-box__heading padding-y-105"
                id="facility-details"
              >
                Facility Details
              </h4>
            </div>
            <span
              aria-hidden="true"
              class="usa-tooltip__body"
              data-testid="tooltipBody"
              id="[object Object]"
              role="tooltip"
              title="Specific healthcare facility where the encounter took place."
            >
              Specific healthcare facility where the encounter took place.
            </span>
          </span>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Facility Name
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  PRM- Palmdale Regional Medical Center
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Facility Address
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  5001 North Mount Washington Circle Drive
North Canton, MA 02740
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Facility Contact Address
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  5001 North Mount Washington Circle Drive
North Canton, MA 02740
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Facility Type
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  Healthcare Provider
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Facility ID
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  2.16.840.1.113883.4.6
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="gutter-3"
        >
          <span
            class="usa-tooltip"
            data-testid="tooltipWrapper"
          >
            <div
              aria-describedby="[object Object]"
              class="usa-tooltip__trigger usa-tooltip short-tooltip"
              data-testid="triggerElement"
              tabindex="0"
              title=""
            >
              <h4
                class="usa-summary-box__heading padding-y-105"
                id="provider-details"
              >
                Provider Details
              </h4>
            </div>
            <span
              aria-hidden="true"
              class="usa-tooltip__body"
              data-testid="tooltipBody"
              id="[object Object]"
              role="tooltip"
              title="Person or organization that provided care during the encounter."
            >
              Person or organization that provided care during the encounter.
            </span>
          </span>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Provider Name
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  test provider name
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Provider Contact
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  test provider contact
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
